tools/shared/src/main/scala/org/scalajs/core/tools/javascript/Printers.scala

```diff
@@ -80,180 +89,253 @@ object Printers {
     def printTree(tree: Tree, isStat: Boolean): Unit = {
       tree match {
         case EmptyTree =>
-          print("<empty>")
+          printString("<empty>")
 
         // Comments
 
-        case DocComment(text) =>
-          val lines = text.split("\n").toList
+        case tree: DocComment =>
+          val lines = tree.text.split("\n").toList
```

Comment:
I very much doubt that this change (and similar ones in this file) improves things whatsoever. For case classes, the scalac compiler bypasses the `unapply` methods in pattern matching, and directly goes to `isInstanceOf` + field access.